File: Server/src/logic/player/externalplayer/ExternalBoard.java
package logic.player.externalplayer;

import connection.ClientAction;
import connection.Connection;
import logic.cards.Permanent;
import logic.player.Board;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Aedan Smith.
 */

public class ExternalBoard implements Board {
    private ArrayList<Permanent> permanents = new ArrayList<>();
    private Connection connection;

    public ExternalBoard(Connection connection){
        this.connection = connection;
    }

    @Override
    public void add(Permanent permanent) {
        connection.out.println(ClientAction.PLAY_CARD + permanent.getId());
        permanents.add(permanent);
    }

    @Override
    public String remove(int n) {
        connection.out.println(ClientAction.KILL_CARD + (char) (n+33));
        return permanents.remove(n).getId();
    }

    @Override
    public List<Permanent> getBoard() {
        return permanents;
    }
}

File: Server/src/logic/player/externalplayer/ExternalBoard.kt
package logic.player.externalplayer

import logic.cards.Permanent
import logic.player.Board
import server.Connection
import java.util.*

/**
 * Created by Aedan Smith.
 */

class ExternalBoard(private val connection: Connection) : Board {
    private val permanents = ArrayList<Permanent>()

    override fun add(permanent: Permanent) {
        connection.getOutput().println(commands.getString("PLAY_CARD") + permanent.id)
        permanents.add(permanent)
    }

    override fun remove(n: Int): String {
        connection.getOutput().println(commands.getString("KILL_CARD") + n)
        return permanents.removeAt(n).id
    }

    override val board = permanents
}
